packages/atomic/src/components/commerce/atomic-commerce-recommendation-list/atomic-commerce-recommendation-list.tsx

```diff
@@ -359,20 +343,32 @@ export class AtomicCommerceRecommendationList
   }
 
   private renderAsGrid(product: Product, i: number) {
-    const atomicProductProps = this.getAtomicProductProps(product);
+    const propsForAtomicProduct = this.getAtomicProductProps(product);
+    const {interactiveProduct} = propsForAtomicProduct;
     return (
       <DisplayGrid
         item={{
           ...product,
           clickUri: product.clickUri,
           title: product.ec_name ?? '',
         }}
-        {...atomicProductProps.interactiveProduct}
+        select={() => {
+          this.logWarningIfNeeded(interactiveProduct.warningMessage);
+          interactiveProduct.select();
+        }}
+        beginDelayedSelect={() => {
+          this.logWarningIfNeeded(interactiveProduct.warningMessage);
+          interactiveProduct.beginDelayedSelect();
+        }}
+        cancelPendingSelect={() => {
+          this.logWarningIfNeeded(interactiveProduct.warningMessage);
+          interactiveProduct.cancelPendingSelect();
+        }}
         setRef={(element) =>
           element && this.itemListCommon.setNewResultRef(element, i)
         }
       >
-        <atomic-product {...atomicProductProps}></atomic-product>
+        <atomic-product {...propsForAtomicProduct}></atomic-product>
```

Comment:
is InteractiveProduct being used in the atomic-product?

If not, see my suggestion above L346-347.
If yes, dismiss it